### PR TITLE
improved FindMLPACK.cmake

### PR DESCRIPTION
**Pull request by flying-sheep**

### CODE REVIEW

File: CMake/FindMLPACK.cmake [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	PATHS "$ENV{ProgramFiles}/mlpack"
+)
+
+
+if(MLPACK_INCLUDE_DIR)
+	# Read and parse mlpack version header file for version number
+	file(STRINGS "${MLPACK_INCLUDE_DIR}/mlpack/core/util/version.hpp" _mlpack_HEADER_CONTENTS REGEX "#define MLPACK_VERSION_[A-Z]+ ")
+	string(REGEX REPLACE ".*#define MLPACK_VERSION_MAJOR ([0-9]+).*" "\\1" MLPACK_VERSION_MAJOR "${_mlpack_HEADER_CONTENTS}")
+	string(REGEX REPLACE ".*#define MLPACK_VERSION_MINOR ([0-9]+).*" "\\1" MLPACK_VERSION_MINOR "${_mlpack_HEADER_CONTENTS}")
+	string(REGEX REPLACE ".*#define MLPACK_VERSION_PATCH ([0-9]+).*" "\\1" MLPACK_VERSION_PATCH "${_mlpack_HEADER_CONTENTS}")
```
Comment (zoq) [inline CMake/FindMLPACK.cmake]: If someone uses the git repository, the version would look something like:

```
#define __MLPACK_VERSION_MAJOR 3
#define __MLPACK_VERSION_MINOR 0
#define __MLPACK_VERSION_PATCH "x"
```

Do you think you can adapt the regex accordingly?

File: CMake/FindMLPACK.cmake [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#   MLPACK_VERSION_MAJOR - major version number
+#   MLPACK_VERSION_MINOR - minor version number
+#   MLPACK_VERSION_PATCH - patch version number
+#   MLPACK_VERSION_STRING - version number as a string (ex: "1.0.4")
+
+
+# UNIX paths are standard, no need to specify them.
+find_library(MLPACK_LIBRARY
+	NAMES mlpack
+	PATHS "$ENV{ProgramFiles}/mlpack/lib"  "$ENV{ProgramFiles}/mlpack/lib64" "$ENV{ProgramFiles}/mlpack"
```
Comment (zoq) [inline CMake/FindMLPACK.cmake]: Should we include: ``/usr/lib64/ /usr/lib/ /usr/local/lib64/ /usr/local/``?

Comment (flying-sheep) [reply CMake/FindMLPACK.cmake]: No: “UNIX paths are standard, no need to specify them”

File: CMake/FindMLPACK.cmake (hunk 1 above)
Comment (flying-sheep) [reply CMake/FindMLPACK.cmake]: Sure, I’ll do it once I get to it.

Comment (zoq) [reply CMake/FindMLPACK.cmake]: Thanks!

File: CMake/FindMLPACK.cmake (hunk 2 above)
Comment (zoq) [reply CMake/FindMLPACK.cmake]: Thanks for the clarification, somehow missed the comment.

**Comment by flying-sheep** [comment]
OK, done!

**Comment by zoq** [comment]
Thanks again for the great contribution.

**Pull request merged**